cscs-checks/apps/cpmd/cpmd_check.py

```diff
@@ -7,47 +7,32 @@
 import reframe.utility.sanity as sn
 
 
-@rfm.parameterized_test(['small'], ['large'])
+@rfm.simple_test
 class CPMDCheck(rfm.RunOnlyRegressionTest):
-    def __init__(self, scale):
-        self.descr = 'CPMD check (C4H6 metadynamics)'
-        self.maintainers = ['AJ', 'LM']
-        self.tags = {'production'}
+    scale = parameter(['small', 'large'])
+    descr = 'CPMD check (C4H6 metadynamics)'
+    maintainers = ['AJ', 'LM']
+    tags = {'production'}
+    valid_systems = ['daint:gpu']
+    num_tasks_per_node = 1
+    valid_prog_environs = ['builtin']
+    modules = ['CPMD']
+    executable = 'cpmd.x'
+    executable_opts = ['ana_c4h6.in > stdout.txt']
+    readonly_files = ['ana_c4h6.in', 'C_MT_BLYP', 'H_MT_BLYP']
+    use_multithreading = True
+    strict_check = False
+    extra_resources = {
+        'switches': {
+            'num_switches': 1
+        }
+    }
 
-        self.valid_systems = ['daint:gpu']
-        if scale == 'small':
+    @run_after('init')
+    def setup_by_scale(self):
+        if self.scale == 'small':
             self.num_tasks = 9
             self.valid_systems += ['dom:gpu']
-        else:
-            self.num_tasks = 16
-
-        self.num_tasks_per_node = 1
-        self.valid_prog_environs = ['builtin']
-        self.modules = ['CPMD']
-        self.executable = 'cpmd.x'
-        self.executable_opts = ['ana_c4h6.in > stdout.txt']
-        self.readonly_files = ['ana_c4h6.in', 'C_MT_BLYP', 'H_MT_BLYP']
-        self.use_multithreading = True
-        self.strict_check = False
-        self.extra_resources = {
-            'switches': {
-                'num_switches': 1
-            }
-        }
-
-        #  OpenMP version of CPMD segfaults
-        #  self.variables = { 'OMP_NUM_THREADS' : '8' }
-        energy = sn.extractsingle(
-            r'CLASSICAL ENERGY\s+-(?P<result>\S+)',
-            'stdout.txt', 'result', float)
-        energy_reference = 25.81
-        energy_diff = sn.abs(energy - energy_reference)
-        self.sanity_patterns = sn.assert_lt(energy_diff, 0.26)
-        self.perf_patterns = {
-            'time': sn.extractsingle(r'^ cpmd(\s+[\d\.]+){3}\s+(?P<perf>\S+)',
-                                     'stdout.txt', 'perf', float)
-        }
-        if scale == 'small':
             self.reference = {
```

Comment:
I would rather add a reference dictionary at the class level with all the references and select here the valid set of references based on the variant. See how we've done it in other tests with `allref`.